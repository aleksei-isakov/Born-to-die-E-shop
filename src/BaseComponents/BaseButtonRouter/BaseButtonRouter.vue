<template>
  <button class="BaseButtonRouter" @click="click">
    <router-link :to="require(path)"><slot></slot></router-link>
  </button>
</template>

<script>
export default {
  name: 'BaseButtonRouter',

  props: {
    path: {
      type: String,
      default: ''
    },

    data() {
      return {};
    },

    methods: {
      click() {
        this.$emit('click', data);
      }
    }
  }
};
</script>
