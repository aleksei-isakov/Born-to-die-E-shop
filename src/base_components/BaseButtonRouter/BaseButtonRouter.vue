<template>
  <router-link :to="path">
    <slot></slot>
  </router-link>
</template>

<script>
export default {
  name: 'BaseButtonRouter',

  props: {
    path: {
      type: String,
      default: '',
      required: true
    }
  }
};
</script>

<style lang="scss" scoped>
@import '@/scss/variables.scss';
.base-button-router {
  display: block;
  text-transform: uppercase;
  color: #e7e7e7;
  font-weight: 500;
  font-size: 16px;
  line-height: 22px;
  letter-spacing: 0.03em;
  font-family: Roboto, sans-serif;
  transition: $transition;
  &:hover {
    text-decoration: none;
    color: $white;
  }
  &.router-link-exact-active {
    color: $white;
  }
}
</style>
