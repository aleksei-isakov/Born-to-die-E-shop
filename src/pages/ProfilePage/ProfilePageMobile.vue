<template>
  <div class="profile-page__container">
    <div class="profile-page__wrapper">
      <div class="profile-page__header">
        <h1>Profile</h1>
      </div>
      <div class="profile-page__sections">
        <div class="profile-page__sidebar">
          <div class="profile-page__sidebar_info">
            <div class="profile-page__sidebar_item">
              <base-button-router path="/profile">
                <i class="fas fa-house-user fa-2x"></i>
                Profile
              </base-button-router>
            </div>
            <div class="profile-page__sidebar_item">
              <base-button-router path="/wishlist">
                <i class="fas fa-heart fa-2x"></i>
                Wishlist
              </base-button-router>
            </div>
            <div class="profile-page__sidebar_item">
              <base-button-router path="/orders">
                <i class="fas fa-folder fa-2x"></i>
                Orders
              </base-button-router>
            </div>
            <div class="profile-page__sidebar_item">
              <base-button-router path="/address-book">
                <i class="fas fa-address-book fa-2x"></i>
                Address book
              </base-button-router>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import BaseButtonRouter from '@/base_components/BaseButtonRouter/BaseButtonRouter';
export default {
  name: 'ProfilePageMobile',
  components: {
    BaseButtonRouter
  }
};
</script>

<style lang="scss" scoped>
@import '@/scss/CustomVariables.scss';

@media (max-width: $tablet-size) {
  .profile-page {
    &__header {
      display: flex;
      align-self: flex-start;
      padding: 40px;

      h1 {
        font-weight: normal;
        color: $font-color-subtitle;
      }
    }

    &__sidebar {
      position: absolute;
      bottom: 0;
      display: flex;
      align-items: center;
      justify-content: center;
      width: 100%;
      background: $primary;
      min-height: 80px;
    }

    &__sidebar_info {
      width: 100%;
      display: flex;
      align-items: space-around;
    }

    &__sidebar_item {
      display: flex;
      align-items: center;
      justify-content: center;
      flex-direction: column;
      width: 25%;
      height: 100%;
      i {
        color: #ffffff;
      }
      .base-button-router {
        display: flex;
        color: #ffffff;
        opacity: 50%;
        &:hover {
          text-decoration: none;
          color: #ffffff;
          opacity: 100%;
        }
      }
    }
  }
}
</style>
