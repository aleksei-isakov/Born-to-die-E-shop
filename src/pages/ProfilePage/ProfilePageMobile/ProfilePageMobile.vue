<template>
  <div v-if="isMobile" class="profile-page__container">
    <div class="profile-page__wrapper">
      <div class="profile-page__header">
        <h1>Profile</h1>
      </div>
      <div class="profile-page__sections">
        <div class="profile-page__bottombar">
          <div class="profile-page__bottombar_info">
            <div class="profile-page__bottombar_item">
              <i class="fas fa-house-user fa-2x"></i>
              <base-button-router path="/profile"> Profile </base-button-router>
            </div>
            <div class="profile-page__bottombar_item">
              <i class="fas fa-heart fa-2x"></i>
              <base-button-router path="/wishlist">
                Wishlist
              </base-button-router>
            </div>
            <div class="profile-page__bottombar_item">
              <i class="fas fa-folder fa-2x"></i>
              <base-button-router path="/orders"> Orders </base-button-router>
            </div>
            <div class="profile-page__bottombar_item">
              <i class="fas fa-address-book fa-2x"></i>
              <base-button-router path="/address-book">
                Address book
              </base-button-router>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import BaseButtonRouter from '@/base_components/BaseButtonRouter/BaseButtonRouter';
import { mapGetters } from 'vuex';
export default {
  name: 'ProfilePageMobile',
  components: {
    BaseButtonRouter
  },

  computed: {
    ...mapGetters('ProfilePageModule', ['isDesktop', 'isMobile'])
  }
};
</script>

<style lang="scss" scoped>
@import '@/scss/CustomVariables.scss';

.profile-page {
  &__header {
    display: flex;
    align-self: flex-start;
    padding: 40px;

    h1 {
      font-weight: normal;
      color: $font-color-subtitle;
    }
  }

  &__bottombar {
    position: absolute;
    bottom: 0;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100%;
    background: $primary;
    min-height: 80px;
  }

  &__bottombar_info {
    width: 100%;
    display: flex;
    align-items: space-around;
  }

  &__bottombar_item {
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
    width: 25%;
    height: 100%;
    opacity: 50%;
    i {
      color: #ffffff;
    }
    .base-button-router {
      color: #ffffff;
      text-decoration: none;
      &:hover {
        color: #ffffff;
        opacity: 100%;
      }
    }
    &:hover {
      color: #ffffff;
      opacity: 100%;
    }
  }
}
</style>
