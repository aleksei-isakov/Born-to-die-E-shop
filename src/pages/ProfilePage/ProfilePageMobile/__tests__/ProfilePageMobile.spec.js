import { createLocalVue, shallowMount } from '@vue/test-utils';
import Vuex from 'vuex';
import ProfilePageMobile from '@/pages/ProfilePage/ProfilePageMobile/ProfilePageMobile';
import BaseButtonRouter from '@/base_components/BaseButtonRouter/BaseButtonRouter';

let state;
let wrapper;
let store;
let actions;
let getters;
const localVue = createLocalVue();

localVue.use(Vuex);

beforeEach(() => {
  getters = {
    isDesktop: () => false,
    isMobile: () => true
  };

  store = new Vuex.Store({
    modules: {
      ProfilePageModule: {
        actions,
        getters,
        state,
        namespaced: true
      }
    }
  });

  wrapper = shallowMount(ProfilePageMobile, {
    stubs: {
      BaseButtonRouter: BaseButtonRouter
    },

    store,
    localVue
  });
});

afterEach(() => {
  wrapper.destroy();
});

describe('ProfilePageMobile', () => {
  it('renders the mobile version of ProfilePage', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should contain bottombar', () => {
    expect(wrapper.find('.profile-page__bottombar').exists()).toBe(true);
  });
});
