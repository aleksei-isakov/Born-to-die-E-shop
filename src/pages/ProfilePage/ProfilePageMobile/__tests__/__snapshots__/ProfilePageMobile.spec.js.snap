// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePageMobile renders the mobile version of ProfilePage 1`] = `
<div class="profile-page__container">
  <div class="profile-page__wrapper">
    <div class="profile-page__header">
      <h1>Profile</h1>
    </div>
    <div class="profile-page__sections">
      <div class="profile-page__bottombar">
        <div class="profile-page__bottombar_info">
          <div class="profile-page__bottombar_item"><i class="fas fa-house-user fa-2x"></i>
            <router-link to="/profile" class="base-button-router"> Profile </router-link>
          </div>
          <div class="profile-page__bottombar_item"><i class="fas fa-heart fa-2x"></i>
            <router-link to="/wishlist" class="base-button-router">
              Wishlist
            </router-link>
          </div>
          <div class="profile-page__bottombar_item"><i class="fas fa-folder fa-2x"></i>
            <router-link to="/orders" class="base-button-router"> Orders </router-link>
          </div>
          <div class="profile-page__bottombar_item"><i class="fas fa-address-book fa-2x"></i>
            <router-link to="/address-book" class="base-button-router">
              Address book
            </router-link>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
